plugins/persisted-scope/src/lib.rs

```diff
@@ -66,34 +66,30 @@ fn fix_pattern(ac: &AhoCorasick, s: &str) -> String {
     s
 }
 
+const RESURSIVE_DIRECTORY_SUFFIX: &str = "**";
+const DIRECTORY_SUFFIX: &str = "*";
+
 fn detect_scope_type(scope_state_path: &str) -> TargetType {
-    if scope_state_path.ends_with(format!("{}{}", MAIN_SEPARATOR, "**").as_str()) {
+    if scope_state_path.ends_with(RESURSIVE_DIRECTORY_SUFFIX) {
         TargetType::RecursiveDirectory
-    } else if scope_state_path.ends_with(format!("{}{}", MAIN_SEPARATOR, "*").as_str()) {
+    } else if scope_state_path.ends_with(DIRECTORY_SUFFIX) {
         TargetType::Directory
     } else {
         TargetType::File
     }
 }
 
-fn fix_directory(path: &str) -> &str {
-    if let Some(path) = path.strip_suffix(format!("{}{}", MAIN_SEPARATOR, "*").as_str()) {
-        return fix_directory(path);
-    } else if let Some(path) = path.strip_suffix(format!("{}{}", "*", MAIN_SEPARATOR).as_str()) {
-        return fix_directory(path);
-    }
-
-    path
-}
+fn fix_directory(path_str: &str) -> &str {
+    let mut path = Path::new(path_str);
 
-fn fix_recursive_directory(path: &str) -> &str {
-    if let Some(path) = path.strip_suffix(format!("{}{}", MAIN_SEPARATOR, "**").as_str()) {
-        return fix_recursive_directory(path);
-    } else if let Some(path) = path.strip_suffix(format!("{}{}", "**", MAIN_SEPARATOR).as_str()) {
-        return fix_recursive_directory(path);
+    if path.ends_with(DIRECTORY_SUFFIX) || path.ends_with(RESURSIVE_DIRECTORY_SUFFIX) {
+        path = match path.parent() {
+            Some(value) => value,
+            None => return path_str,
+        };
     }
 
-    path
+    path.as_os_str().to_str().unwrap_or(path_str)
```

Comment:
small nitpick, Path has a `to_str()` method too, no need to get the os str first iirc.